huaweicloud/services/scm/data_source_huaweicloud_scm_certificates.go

```diff
@@ -233,17 +239,32 @@ func flattenListCertificatesBodyCertificates(resp interface{}) []interface{} {
 	return rst
 }
 
+func filterListCertificatesBodyCertificate(all []interface{}, d *schema.ResourceData) []interface{} {
+	rst := make([]interface{}, 0, len(all))
+	for _, v := range all {
+		if param, ok := d.GetOk("name"); ok && param != utils.PathSearch("name", v, nil) {
```

Comment:
The judgment of `param, ok := d.GetOk("name")` can be moved outside the loop

Reply:
it's too difficult to implement, and this's clear now, so keep it as it is.

Reply:
all right